python/tvm/relay/ir_pass.py

```diff
@@ -1,16 +1,369 @@
+                    Module/Function]]
+            The implemented optimization pass.
+
+         Returns
+        -------
+        ret : Pass
+            A module level pass built through pass_func.
+        """
+        if not isinstance(pass_func, (types.FunctionType, types.LambdaType)):
+            raise TypeError("pass_func must be a callable for Module pass")
+
+        return _ir_pass.CreateModulePass(name if name else pass_func.__name__,
+                                         opt_level, required, pass_func)
+
+    return create_module_pass
+
+
+def function_pass(opt_level, name=None, required=None):
+    """Create a function pass.
+
+    Parameters
+    ----------
+    opt_level : int
+        The optimization level of this module pass.
+
+    name : Optional[str]
+        The name of the function pass. The name could be empty. In this case, the
+        name of the optimization function will be used as the pass name.
+
+    required : Optional[List[str]]
+        The list of passes that the module pass is dependent on.
+
+    Returns
+    -------
+    create_module_pass : Callable
+        The callable that will create a module pass.
+
+    Examples
+    --------
+    The following code creates a function level pass that performs constant
+    folding.
+
+    .. code-block:: python
+        @relay.ir_pass.function_pass(opt_level=2)
+        def transform(func, ctx):
+            return ir_pass.fold_constant(func)
+
+        function_pass = transform
+        assert isinstance(function_pass, ir_pass.FunctionPass)
+        assert function_pass.info.opt_level == 2
+
+        # Given a module m, the optimization could be invoked as the follwoing:
+        updated_mod = function_pass(m)
+        # Now constant folding should have been applied to every function in
+        # the provided module m. And the updated module will be returned.
+    """
+
+    required = required if required else []
+    if not isinstance(required, (list, tuple)):
+        raise TypeError("Required is expected to be the type of " +
+                        "list/tuple.")
+
+    def create_function_pass(pass_func):
+        """Create the function pass using the user defined optimization
+        function.
+
+        Parameters
+        ----------
+        pass_func : Optional[Callable[(Module/Function, PassContext) ->
+                    Module/Function]]
+            The implemented optimization pass.
+
+        Returns
+        -------
+        ret : Pass
+            A function level pass built through pass_func.
+        """
+        if not isinstance(pass_func, (types.FunctionType, types.LambdaType)):
+            raise TypeError("pass_func must be a callable for Module pass")
+
+        return _ir_pass.CreateFunctionPass(name if name else pass_func.__name__,
+                                           opt_level, required, pass_func)
+
+    return create_function_pass
+
+
+def sequential_pass(opt_level, passes, name="sequential_pass", required=None,
```

Comment:
To be consistent with other functions, argument order should be:
- passes
- name, can have a default value "sequential_pass"
-  opt_level can have a default value.

Reply:
We can find that this order is in particular inconsistent with others.